sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpReceiver.cs

```diff
@@ -543,10 +545,13 @@ private void ThrowLockLostException()
             IDictionary<string, object> propertiesToModify = null,
             CancellationToken cancellationToken = default) =>
             await _retryPolicy.RunOperation(
-                async (timeout) => await AbandonInternalAsync(
-                    lockToken,
+                async (receiver, lckToken, properties, timeout, _) => await receiver.AbandonInternalAsync(
```

Comment:
nit: mark lambdas static

Reply:
Do we already have CSharp v9 support available?

Wouldn't it much better to just elide the async statemachinery instead?

Reply:
>Do we already have CSharp v9 support available?

I think so.

>Wouldn't it much better to just elide the async statemachinery instead?

I think we can do both.

Reply:
Ok let me know if I can elide the state machine and whether CSharp 9 can be used. Then I will update my local tooling and try to give it a go in a next iteration.

Reply:
Pushed the static lambda changes. Waited with avoiding the async state machine since it looks like the current codebase doesn't really do it, so I want to make sure it is ok to do so before I go down spending more time

Reply:
We don't do it by default only if it's on the hot path of a mainline scenario that often goes async. I think a memory profile or receiving/sending session would be our main guide here.

Reply:
Opted out form async/await where possible